pkg/nsx/services/ipblocksinfo/ipblocksinfo.go

```diff
@@ -392,11 +417,37 @@ func (s *IPBlocksInfoService) getIPBlockCIDRsFromStore(pathSet sets.Set[string],
 			return nil, fmt.Errorf("failed to get IPBlock %s from NSX", path)
 		}
 		ipblock := obj.(*model.IpAddressBlock)
-		if ipblock.Cidr == nil { //nolint:staticcheck //ipblock.Cidr is deprecated
-			return nil, fmt.Errorf("failed to get CIDR from ipblock %s", path)
+		if ipblock.Cidrs != nil {
+			ipCIDRs = append(ipCIDRs, ipblock.Cidrs...)
+			log.V(2).Info("Successfully get cidrs for IPBblock", "path", path, "cidrs", ipblock.Cidrs)
+		} else if ipblock.Cidr != nil { //nolint:staticcheck //ipblock.Cidr is deprecated
+			ipCIDRs = append(ipCIDRs, *ipblock.Cidr)                                                  //nolint:staticcheck //ipblock.Cidr is deprecated
+			log.V(2).Info("Successfully get cidrs for IPBblock", "path", path, "cidrs", ipblock.Cidr) //nolint:staticcheck //ipblock.Cidr is deprecated
+		} else {
+			log.V(1).Info("No CIDRs found for IPBlock", "path", path)
 		}
-		log.V(2).Info("successfully get cidr for IPBblock", "path", path, "cidr", *ipblock.Cidr) //nolint:staticcheck //ipblock.Cidr is deprecated
-		ipCIDRs = append(ipCIDRs, *ipblock.Cidr)                                                 //nolint:staticcheck //ipblock.Cidr is deprecated
 	}
 	return ipCIDRs, nil
 }
+
+func (s *IPBlocksInfoService) getIPBlockRangesFromStore(pathSet sets.Set[string], ipBlockStore *IPBlockStore) ([]v1alpha1.IPPoolRange, error) {
+	ipRanges := []v1alpha1.IPPoolRange{}
+	for path := range pathSet {
+		obj := ipBlockStore.GetByKey(path)
+		if obj == nil {
+			return nil, fmt.Errorf("failed to get IPBlock %s from NSX", path)
+		}
+		ipblock := obj.(*model.IpAddressBlock)
+		if ipblock.Ranges == nil {
```

Comment:
Check len(ipblock.Ranges) > 0 to also include the empty array case?

Reply:
range will skip the empty array case. Add ut for it